### PR TITLE
1072 facebook backend

### PR DESCRIPTION
**Pull request by BenRogersNewsome**

### CODE REVIEW

**Review by pauricthelodger — APPROVED** [review]
LGTM!

File: compliant_social_django/decorators.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+def create_audit_logs(audit_logger: AbstractBaseAuditLogger):
+    """
+    A decorator which injects audit logs into a social_core backend object.
+
+    Attributes:
+        audit_logger    An instantiated instance of an audit logger which inherits from
+                        compliant_social_django.audit.clients.AbstractBaseAuditLogger
+    """
+    def decorator(backend):
```
Comment (ZephSibley) [inline compliant_social_django/decorators.py]: The backend param should also be typed as we're relying on its interface here.

File: compliant_social_django/backends/compliant_facebook.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from social_core.backends.facebook import FacebookOAuth2
+from ..storage import AuditLogger
+from ..decorators import create_audit_logs
+
+
+@create_audit_logs(AuditLogger())
```
Comment (ZephSibley) [inline compliant_social_django/backends/compliant_facebook.py]: Might be neater to initialise the AuditLogger inside the function, not sure though, just seen it done that way typically.

**Review by ZephSibley — CHANGES_REQUESTED** [review]
Truly amazing stuff, just needs tests! 😄

File: compliant_social_django/decorators.py (hunk 1 above)
Comment (BenRogersNewsome) [reply compliant_social_django/decorators.py]: Yea, ill type it to the OAuth2 backend

File: compliant_social_django/backends/compliant_facebook.py (hunk 2 above)
Comment (BenRogersNewsome) [reply compliant_social_django/backends/compliant_facebook.py]: Yea, seems like a better way of doing it

**Review by ZephSibley — APPROVED** [review]

**Pull request merged**